Added New Material System Documentation (#1730)
o3de/o3de.org: content/docs/atom-guide/look-dev/materials/material-file-spec.md

toc: true
---

## Elements

Material files (`*.material`) are written in JSON format and contain the following elements.

### **description** (*optional*)  
Provides a description or comment from the material's author.

> ### **materialType**
> The path to the material type. Materials must reference a material type which provides the list of available properties and the shaders to use for rendering. The path must be relative to the asset root or to the material file.

Reviewer: ```suggestion
### **materialType**
The path to the material type. Materials must reference a material type that provides the list of properties and shaders to use for rendering. The path must be relative to the asset root or to the material file.
```

Author: I think my original text is more grammatically correct and accurate.

> ### **materialTypeVersion** (*optional*)  
> Indicates the version number of the material type that was used to create this material. This can be used by the material type version update feature to support backward compatibility with older versions of the material type. (See [Material Type Specification](material-type-file-spec)'s [`version`](material-type-file-spec/#version) and [`versionUpdates`](material-type-file-spec/#versionupdates)).

Reviewer: ```suggestion
### **materialTypeVersion** (*optional*)  
Indicates the version number of the material type that was used to create this material. This can be used by the material type version update feature to support backward compatibility with older versions of the material type. (See [`version`](material-type-file-spec/#version) and [`versionUpdates`](material-type-file-spec/#versionupdates)) in the Material Type File Specification.
```

### **parentMaterial** (*optional*)  

> The path to the parent material file. If specified, the material inherits the properties of another parent material. The parent material must have the same material type as the material. The path must be relative to the asset root or to the material file. If not specified, the material inherits default values directly from the materialType. 

Reviewer: ```suggestion
The path to the parent material file. If specified, the material inherits the properties of another parent material. The parent material must have the same material type as the material. The path must be relative to the asset root or to the material file. If not specified, the material inherits default values directly from the material type.
. 
```

### **propertyValues**

> Lists the names and values to set for material properties. Any properties that aren't specified here will inherit the value from the parent material if available or from the material type's default value. The JSON data type used for the values should match the property's data type as specified in the .materialtype file. However, some data type conversion is supported (i.e. "1" can be used to set a float property to "1.0", etc.).

Reviewer: Per Style Guide, replacing abbreviations and Latin phrases for full English words. I don't think it's needed, but you can add "and so on", or similar, to the end of the example if you'd like. 
```suggestion
Lists the names and values to set for material properties. Any properties that aren't specified here will inherit the value from the parent material if available or from the material type's default value. The JSON data type used for the values should match the property's data type as specified in the `.materialtype` file. However, some data type conversion is supported (for example, "1" can be used to set a float property to "1.0").
```

Image properties must have a path to the source image file. The path must be relative to the asset root or to the material file.

Enum properties must use one of the available enum values as specified by the property definition in the .materialtype file.

All other property types must be specified according to their standard JSON serialization, see [JSON Serialization of O3DE Data Types](/docs/user-guide/programming/serialization/json-data-types).

The example below demonstrates how to set a property of each possible data type.

### **properties** (deprecated)

> An older version of the material file format organized properties by group in nested JSON objects. This has been replaced by `propertyValues` (above) which uses a flat list of property IDs. The engine will still load the old format but any new material files should use the new format.

Reviewer: Swap "should" with "must", as "should" indicates something is highly suggested, but optional. If that's intended, then disregard this. 

```suggestion
An older version of the material file format organized properties by group in nested JSON objects. This is replaced by `propertyValues` (above), which uses a flat list of property identifiers. The engine can still load the old format, but new material files must use the new format.
```

## Example

```json
{

>     "description":  "This example uses a pretend material type, parent, and properties that don't actually exist.",

Reviewer: ```suggestion
    "description":  "This is an example only. The material type, parent, and properties don't exist.",
```

### **description** (*optional*)  
Provides a description or comment from the material's author.

### **materialType**
The path to the material type. Materials must reference a material type, which provides the list of available properties and the shaders to use for rendering. The path must be relative to the asset root or to the material file.

### **materialTypeVersion** (*optional*)  
Indicates the version number of the material type that was used to create this material. This can be used by the material type version update feature to support backward compatibility with older versions of the material type. (See [`version`](material-type-file-spec/#version) and [`versionUpdates`](material-type-file-spec/#versionupdates)) in the Material Type File Specification.

### **parentMaterial** (*optional*)  
The path to the parent material file. If specified, the material inherits the properties of another parent material. The parent material must have the same material type as the material. The path must be relative to the asset root or to the material file. If not specified, the material inherits default values directly from the material type.

> . 

Reviewer: ```suggestion

```

### **propertyValues**
Lists the names and values to set for material properties. Any properties that aren't specified here will inherit the value from the parent material if available or from the material type's default value. The JSON data type used for the values should match the property's data type as specified in the `.materialtype` file. However, some data type conversion is supported (for example, "1" can be used to set a float property to "1.0").

Image properties must have a path to the source image file. The path must be relative to the asset root or to the material file.

> Enum properties must use one of the available enum values as specified by the property definition in the .materialtype file.

Reviewer: ```suggestion
Enum properties must use one of the available enum values as specified by the property definition in the `.materialtype` file.
```
